src/core/ext/client_config/client_channel.c

```diff
@@ -577,11 +607,15 @@ static void continue_picking(grpc_exec_ctx *exec_ctx, void *arg,
     /* cancelled, do nothing */
   } else if (error != GRPC_ERROR_NONE) {
     grpc_exec_ctx_sched(exec_ctx, cpa->on_ready, GRPC_ERROR_REF(error), NULL);
-  } else if (pick_subchannel(exec_ctx, cpa->elem, cpa->initial_metadata,
-                             cpa->initial_metadata_flags,
-                             cpa->connected_subchannel, cpa->on_ready,
-                             GRPC_ERROR_NONE)) {
-    grpc_exec_ctx_sched(exec_ctx, cpa->on_ready, GRPC_ERROR_NONE, NULL);
+  } else {
+    call_data *calld = cpa->elem->call_data;
+    gpr_mu_lock(&calld->mu);
```

Comment:
I didn't express myself clearly: my concern was recursive attempts to acquire the lock, given that we schedule another callback while holding the lock. BUT, answering my own question, using `grpc_exec_ctx_sched` to do so is safe (unlike, for example, `grpc_closure_run`). Please disregard my original comment.